docs/reference/mapping/types/json.asciidoc

```diff
@@ -0,0 +1,173 @@
+[[json]]
+=== JSON datatype
+
+experimental[The `json` field type is experimental and may be changed in a breaking way in future releases.]
+
+The `json` mapping type allows an entire JSON object to be indexed as a single field, with support for simple
+queries on the field's contents. A `json` field can be useful for modelling objects with a large number of
+distinct subfields whose keys are not known in advance. This type of situation might arise arise when
+some data is generated directly by users, or by an external third-party application.
```

Comment:
I think it would be useful to explain a bit more about the trade offs of using this vs explicitly mapping the object. For example one advantage of this is that we expect the resulting index to be smaller with the disadvantage of limited query support and no support for aggregations and sorting.

Reply:
Makes a lot of sense, I'll expand this section.